bin/sozo/src/ops/events.rs

```diff
@@ -38,126 +37,255 @@ pub async fn execute(
 
+    //print_events_map(events_map.clone());
+
     if let Some(events_map) = events_map {
         parse_and_print_events(res, events_map)?;
     } else {
-        println!("{}", serde_json::to_string_pretty(&res)?);
+        println!("[serde_json] {}", serde_json::to_string_pretty(&res)?);
     }
 
     Ok(())
 }
 
 fn parse_and_print_events(
     res: starknet::core::types::EventsPage,
-    events_map: HashMap<String, Vec<Event>>,
+    events_map: HashMap<String, Vec<Token>>,
 ) -> Result<()> {
     println!("Continuation token: {:?}", res.continuation_token);
     println!("----------------------------------------------");
     for event in res.events {
+        //println!("----------------------------------------------");
+        //println!("[parse_and_print_events]");
+        //println!("{}", serde_json::to_string_pretty(&event)?);
         if let Some(e) = parse_event(event.clone(), &events_map) {
             println!("{e}");
         } else {
-            // Couldn't parse event
-            println!("{}", serde_json::to_string_pretty(&event)?);
+            println!("Couldn't parse event - {}", serde_json::to_string_pretty(&event)?);
+            println!("-----> {}", event.keys[0].to_string());
         }
     }
     Ok(())
 }
 
+fn print_events_map(events_map: Option<HashMap<String, Vec<Token>>>) {
+    match events_map {
+        Some(map) => {
+            for (key, events) in map {
+                println!("[print_events_map] Key: {}", key);
+                for event in events {
+                    // Using {:?} to print the Debug representation of Event
+                    // Ensure that Event implements Debug trait
+                    println!("[print_events_map] {:?}", event);
+                    println!("");
+                }
+                println!("----------------------------------------------");
+            }
+        }
+        None => println!("No events map."),
+    }
+}
+
+fn parse_core_basic(
+    cb: &CoreBasic,
+    value: &FieldElement,
+    is_nested: bool,
+) -> Result<String, String> {
+    match cb.type_name().as_str() {
+        "felt252" => {
+            if is_nested {
+                Ok(format!("\"{:#x}\"", value))
+            } else {
+                match parse_cairo_short_string(value) {
+                    Ok(parsed) => Ok(parsed),
+                    Err(_) => Ok(format!("\"{:#x}\"", value)),
+                }
+            }
+        }
+        "bool" => {
+            if *value == FieldElement::ZERO {
+                Ok("false".to_string())
+            } else {
+                Ok("true".to_string())
+            }
+        }
+        "ClassHash" | "ContractAddress" => Ok(format!("{:#x}", value)),
+        "u8" | "u16" | "u32" | "u64" | "u128" | "usize" | "i8" | "i16" | "i32" | "i64" | "i128" => {
+            Ok(value.to_string())
+        }
+        _ => Err(format!("Unsupported CoreBasic type: {}", cb.type_name())),
+    }
+}
```

Comment:
Would be amazing to have that into cainome, having a `print_value` method on the `CoreBasic` token type.

To allow some configuration, I think we should  have 2 types or choices (mostly for felts and integers) here:
1. Do we want values in hex or decimal.
2. If it's a felt, do we want to add the string representation of it (even if we're inside an array).

This can be done later, just wanted to point out that it's concerning only cainome logic so, may be better to move it there. :+1: